File: Prototype/src/main/java/me/danetnaverno/editoni/minecraft/world/MinecraftBlock.java
package me.danetnaverno.editoni.minecraft.world;

import me.danetnaverno.editoni.common.world.Block;
import net.querz.nbt.CompoundTag;

public class MinecraftBlock extends Block
{
    public MinecraftBlock(MinecraftChunk chunk, int x, int y, int z, CompoundTag tag)
    {
        super(chunk, x, y, z);
        try
        {
            setId(tag.getString("Name"));
        }
        catch (Throwable ex)
        {
            ex.printStackTrace();
        }
    }
}

File: Prototype/src/main/java/me/danetnaverno/editoni/minecraft/world/MinecraftBlock.kt
package me.danetnaverno.editoni.minecraft.world

import me.danetnaverno.editoni.common.blocktype.BlockType
import me.danetnaverno.editoni.common.world.*
import me.danetnaverno.editoni.util.globalToLocalPos
import org.joml.Vector3i

class MinecraftBlock(chunk: Chunk, localPos: Vector3i, blockType: BlockType, blockState: BlockState?, tileEntity: TileEntity?)
    : Block(chunk, localPos, blockType, blockState, tileEntity)
{
    constructor(world: World, globalPos: Vector3i, blockType: BlockType, blockState: BlockState?, tileEntity: TileEntity?)
        : this(world.getChunkByBlockCoord(globalPos.x, globalPos.z), globalToLocalPos(globalPos), blockType, blockState, tileEntity)
}